sdk/include/opentelemetry/sdk/common/http_client.h

```diff
@@ -0,0 +1,129 @@
+#pragma once
+
+#include <chrono>
+#include <string>
+#include <vector>
+#include "opentelemetry/nostd/function_ref.h"
+#include "opentelemetry/version.h"
+
+// Usage Example
+//
+// struct ReponseHandler: public HttpResponseHandler {
+//      void OnHttpResponse(HttpResponse& res) noexcept override
+//      {
+//           if (res.IsSuccess())
+//              .. process response
+//      }
+//  };
+//
+//   HttpSessionManager sessionManager; // implementer can provide singleton implementation for it
+//   auto session = sessionManager.createHttpSession("localhost", 8000);
+//   auto request = session->CreateRequest();
+//   request->AddHeader(..);
+//   ResponseHandler res_handler;
+//   session->SendRequest(res_handler);
+//   session->FinishSession() // optionally in the end.
+//   ...shutdown
+//    sessionManager.FinishAllSessions()
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace common
+{
+namespace http
+{
+
+enum class HttpMethod
+{
+  HTTP_GET,
+  HTTP_POST,
+  HTTP_PUT,
+  HTTP_OPTIONS,
+  HTTP_HEAD,
+  HTTP_PATCH,
+  HTTP_DELETE
+};
+
+enum class HttpSessionState
+{
+  CREATED,  // session object is created
+  ONGOING,  // http(s) request is ongoing
+  FINISHED  // https(s) request is finished
+};
+
+using HttpHeader     = std::pair<std::string, std::string>;
+using HttpBody       = std::vector<uint8_t>;
+using HttpStatusCode = uint16_t;
```

Comment:
Minor: consider enum?

Reply:
enum for status code ? I thought list would be too put in header, and didn't see advantage of managing it as part of API. Let me know if you feel it needs to be there.